files/check_keepalive.rb

```diff
@@ -0,0 +1,97 @@
+module Sensu
+module Extension
+
+# 1) update per-check timestamps in redis
+# 2) have eventmachine periodic timer to check if anything timed out and
+#    generate a new "critical" event for every
+#
+# extension has access to:
+#  - logger     @logger
+#  - redis      @settings[:redis]
+#  - transport  @settings[:transport]
+
+class CheckKeepalive < Handler
+  def name
+    'check_keepalive'
+  end
+
+  def description
+    'emit per-check keepalive events'
+  end
+
+  def post_init
+    @redis    = @settings.redis
+    @em_timer = EventMachine::PeriodicTimer.new(10) { process_keepalives }
+  end
+
+  def stop
+    @em_timer.cancel
+    yield
+  end
+
+  def run(event)
+    client_name = event[:client][:name]
+    check_name  = event[:check][:name]
+
+    rkey = "ext:check_keepalive:checks"
```

Comment:
followup to https://github.com/Yelp/puppet-monitoring_check/pull/33#discussion_r24506313

@solarkennedy Hehe, that is what sensu does (with client keepalives), but we need something that sends the events. That is what this extension needs to do. We can't just send 1 event. This, just like determine_stale_clients, needs to keep sending CRIT events until the watchdog is reset.

you're right, but i think sensu is doing poor job pretending to be state machine - single event should be enough to react and transition. anyway we can't change this so no need to discuss further?

original question:

@solarkennedy I was thinking that if this was stored in the stash namespace, it would be easy to remove. Otherwise, how will we delete these checks when they are no longer needed?

for checks scheduled by sensu-server - i imagine they are deregistered from redis when they are removed and so we can have additional redis lookup for existence of the check.

for standalone checks as i mentioned in previous comment - we either need a hook attached to ensure => absent or we do that manually. how do we do it right now anyway? when check is commented out/removed from puppet repo - what exactly happens to it on sensu-server?